src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -326,8 +328,8 @@ internal ChangeSignatureResult ChangeSignatureWithContext(ChangeSignatureAnalyze
             var updatedRoots = new Dictionary<DocumentId, SyntaxNode>();
             foreach (var docId in nodesToUpdate.Keys)
             {
-                var doc = updatedSolution.GetDocument(docId);
-                var updater = doc.Project.LanguageServices.GetService<AbstractChangeSignatureService>();
+                var doc = originalSolution.GetDocument(docId)!;
```

Comment:
let's add a RequireDocument.  for code that expects this to return non-null (which this code does, it will be much nicer than having to suppress warnings).

Reply:
@jasonmalinowski is this something you have added already?

Reply:
I haven't added it, but yes a GetRequiredDocument that throws would be fantastic. I'll be honest: after doing the nullable stuff my biggest regret is we made GetDocument return null. It made the rare case (where you could have called ContainsDocument first!) easy and the common case hard.